global_helpers/panther_audit.py

```diff
@@ -0,0 +1,35 @@
+import boto3
+import os
+
+PANTHER_MASTER_REGION = os.environ['AWS_REGION']
+
+
+def build_client(resource, service, region=None):
+    """Function builds resource client that assumes Panther audit role"""
+    account = resource['AccountId']
+    role_arn = f'arn:aws:iam::{account}:role/PantherAuditRole-{PANTHER_MASTER_REGION}'
+    sts_connection = boto3.client('sts')
+    acct_b = sts_connection.assume_role(
+        RoleArn=role_arn,
+        RoleSessionName="lambda_assume_audit_role"
+    )
+    ACCESS_KEY = acct_b['Credentials']['AccessKeyId']
+    SECRET_KEY = acct_b['Credentials']['SecretAccessKey']
+    SESSION_TOKEN = acct_b['Credentials']['SessionToken']
```

Comment:
```suggestion
    ACCESS_KEY = acct_b.get('Credentials', {}).get('AccessKeyId', 'MISSING ACCESS KEY')
    SECRET_KEY = acct_b.get('Credentials', {}).get('SecretAccessKey', 'MISSING SECRET ACCESS KEY')
    SESSION_TOKEN = acct_b.get('Credentials', {}).get('SessionToken', 'MISSING SESSION TOKEN')
```

Reply:
Use DeepGet instead of multiple gets!!!

https://github.com/panther-labs/panther-analysis/blob/master/global_helpers/panther_base_helpers.py#L147-L154

Reply:
This is for my panther_audit global module those creds come from the boto3 sts assume role call, there is really no need to use deep dive or gets. If the call fails then we will never get to the part of assigning the keys and tokens. This part is straight from AWS Docs, I can change it, but I don't think its needed.

Reply:
I'm going to say this when I see multiple .get calls